store/sqlstore/group_supplier.go

```diff
@@ -0,0 +1,877 @@
+		if channelResult.Err != nil {
+			result.Err = channelResult.Err
+			return result
+		}
+
+		groupChannel := &GroupChannel{
+			GroupSyncable: *groupSyncable,
+			ChannelId:     groupSyncable.SyncableId,
+		}
+		err = s.GetMaster().Insert(groupChannel)
+	default:
+		result.Err = model.NewAppError("SqlGroupStore.GroupCreateGroupSyncable", "model.group.type.app_error", nil, "group_id="+groupSyncable.GroupId+", syncable_id="+groupSyncable.SyncableId+", "+err.Error(), http.StatusInternalServerError)
+		return result
+	}
+	if err != nil {
+		if err == sql.ErrNoRows {
+			result.Err = model.NewAppError("SqlGroupStore.GroupCreateGroupSyncable", "store.sql_group.no_rows_affected", nil, "group_id="+groupSyncable.GroupId+", syncable_id="+groupSyncable.SyncableId, http.StatusInternalServerError)
+		}
+		result.Err = model.NewAppError("SqlGroupStore.GroupCreateGroupSyncable", "store.insert_error", nil, "group_id="+groupSyncable.GroupId+", syncable_id="+groupSyncable.SyncableId+", "+err.Error(), http.StatusInternalServerError)
+		return result
+	}
+
+	result.Data = groupSyncable
+	return result
+}
+
+func (s *SqlSupplier) GroupGetGroupSyncable(ctx context.Context, groupID string, syncableID string, syncableType model.GroupSyncableType, hints ...store.LayeredStoreHint) *store.LayeredStoreSupplierResult {
+	result := store.NewSupplierResult()
+
+	groupSyncable, err := s.getGroupSyncable(groupID, syncableID, syncableType)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			result.Err = model.NewAppError("SqlGroupStore.GroupGetGroupSyncable", "store.sql_group.no_rows", nil, err.Error(), http.StatusNotFound)
+		} else {
+			result.Err = model.NewAppError("SqlGroupStore.GroupGetGroupSyncable", "store.select_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+		return result
+	}
+
+	result.Data = groupSyncable
+
+	return result
+}
+
+func (s *SqlSupplier) getGroupSyncable(groupID string, syncableID string, syncableType model.GroupSyncableType) (*model.GroupSyncable, error) {
+	var err error
+	var result interface{}
+
+	switch syncableType {
+	case model.GroupSyncableTypeTeam:
+		result, err = s.GetMaster().Get(GroupTeam{}, groupID, syncableID)
+	case model.GroupSyncableTypeChannel:
+		result, err = s.GetMaster().Get(GroupChannel{}, groupID, syncableID)
+	default:
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if result == nil {
+		return nil, sql.ErrNoRows
+	}
+
+	groupSyncable := model.GroupSyncable{}
+	switch syncableType {
+	case model.GroupSyncableTypeTeam:
+		groupTeam := result.(*GroupTeam)
+		groupSyncable.SyncableId = groupTeam.TeamId
+		groupSyncable.GroupId = groupTeam.GroupId
+		groupSyncable.CanLeave = groupTeam.CanLeave
+		groupSyncable.AutoAdd = groupTeam.AutoAdd
+		groupSyncable.CreateAt = groupTeam.CreateAt
+		groupSyncable.DeleteAt = groupTeam.DeleteAt
+		groupSyncable.UpdateAt = groupTeam.UpdateAt
+		groupSyncable.Type = syncableType
+	case model.GroupSyncableTypeChannel:
```

Comment:
It looks like only part of the `GroupSyncable` is being populated as compared to what's done in `GroupGetAllGroupSyncablesByGroup` below. Not sure I understand the difference.

Reply:
You're right the `TeamDisplayName` and `TeamType` aren't being joined in on the single request. These were properties that were needed by the front-end because of the whole "implicit GroupTeam" scenario. The values don't seem to be necessary in the get group syncable single retrieval (yet).